### PR TITLE
[ROCm] upgrade CI to 6.0

### PR DESCRIPTION
**Pull request by jeffdaily**
DO NOT MERGE

cc @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Comment by johnnynunez** [comment]
when?

File: test/test_cuda.py [hunk 1]
```diff
@@ -3417,6 +3417,7 @@ def test_matmul_device_mismatch(self):
                     torch.addmm(s, m1, m2)
 
     @unittest.skipIf(TEST_MULTIGPU, "Testing on one GPU is sufficient")
+    @skipIfRocm(msg="Skipped!!, since it is failing on ROCm 6.0")
```
Comment (malfet) [inline test/test_cuda.py:3420]: This sounds like a regression which will affect distributed workflows pretty badly, but up to you guys

**Review by malfet — APPROVED** [review]
LGTM, though fact that lazy_init not longer works might negatively affect distributed workflows on ROCm platform. 
cc: @wconstab

**Comment by jithunnair-amd** [comment]
@jeffdaily We will also want to include `inductor.yml` changes to ROCm6.0 in this PR to keep it all together. There's a PR to add that workflow here: https://github.com/pytorch/pytorch/pull/110544

**Review by jithunnair-amd — APPROVED** [review]

**Review by pruthvistony — APPROVED** [review]
MIOpen workaround added for the DB permission problem.

**Pull request closed without merging**